app/notifications/validators.py

```diff
@@ -45,14 +48,32 @@
 
 
 def check_service_over_api_rate_limit(service, key_type):
-    if current_app.config["API_RATE_LIMIT_ENABLED"] and current_app.config["REDIS_ENABLED"]:
-        cache_key = rate_limit_cache_key(service.id, key_type)
-        rate_limit = service.rate_limit
-        interval = 60
-        with REDIS_EXCEEDED_RATE_LIMIT_DURATION_SECONDS.time():
-            if redis_store.exceeded_rate_limit(cache_key, rate_limit, interval):
-                current_app.logger.info("service %s has been rate limited for throughput", service.id)
-                raise RateLimitError(rate_limit, interval, key_type)
+    if not current_app.config["API_RATE_LIMIT_ENABLED"]:
+        return
+    if not current_app.config["REDIS_ENABLED"]:
+        return
+
+    with REDIS_EXCEEDED_RATE_LIMIT_DURATION_SECONDS.time():
+        if rate_limit_exceeded(service, key_type):
+            current_app.logger.info("service %s has been rate limited for throughput", service.id)
+            raise RateLimitError(service.rate_limit, SECONDS_IN_1_MINUTE, key_type)
+
+
+def rate_limit_exceeded(service, key_type):
+    cache_key = rate_limit_cache_key(service.id, key_type)
+
+    if service.has_permission("token_bucket"):
```

Comment:
This returns null of it's not in the table, yes? (vs an error)

Reply:
Yes: https://github.com/alphagov/notifications-api/blob/f460b181ae2ee3e938ce7ebc189ea95545c4eede/app/serialised_models.py#L128-L129